### ISSUE TITLE
[BUG]Error with memory=True using AzureOpenAI Provider

### ISSUE THREAD

**Issue by morhyak** [crewAIInc/crewAI#1368]
### Description

Hi,
I encountered an issue when using the AzureOpenAI provider. Setting memory=True results in a parameters error because it doesn't allow defining the embedder with additional parameters beyond model_name and provider.
In my configuration for Azure embeddings, I need to include all parameters, including SSO access details. For example, the setup requires something like the following configuration:
```
"model": EMBEDDING_MODEL,
            "deployment": EMBEDDING_MODEL ,
            "openai_api_key":API_KEY,
            "azure_endpoint": API_BASE,
            "openai_api_version":API_VERSION,
            "openai_api_type":TYPE,
            "default_headers": {
                "Authorization": f"Bearer {access_token}",
                "Content-Type": "application/json"
```
Please let me know if there's a workaround or fix for this issue.
Thanks

### Steps to Reproduce

To reproduce the issue:
crew = Crew(
    agents=[support_agent, support_quality_assurance_agent],
    tasks=[inquiry_resolution, quality_assurance_review],
    verbose=True,
    memory=True,
    llm=llm,
    embedder={
        "provider": "azure_openai",
        "config": {
            "model": EMBEDDING_MODEL,
            "deployment": EMBEDDING_MODEL ,
            "openai_api_key":API_KEY,
            "azure_endpoint": API_BASE,
            "openai_api_version":API_VERSION,
            "openai_api_type":TYPE,
            "default_headers": {
                "Authorization": f"Bearer {access_token}",
                "Content-Type": "application/json"
            }
        }})

### Expected behavior

I expected then to run successfully the crew built(using crew.kickoff)

### Screenshots/Code snippets

-----------------------

### Operating System

Windows 10

### Python Version

3.11

### crewAI Version

0.55.2

### crewAI Tools Version

0.12.1

### Virtual Environment

Venv

### Evidence

SchemaError: Key 'embedder' error:
Key 'config' error:
Wrong key 'azure_deployment' in {'model': 'text-embedding-ada-002', 'azure_deployment': 'text-embedding-ada-002', 'openai_api_key': '1a3dfe7684804aa28f492105d585f950', 'azure_endpoint': 'https://davinci-dev-openai-api.hpcloud.hp.com/indibot', 'openai_api_version': '2023-03-15-preview', 'openai_api_type': 'azure', 'default_headers': {'Authorization': 'Bearer 

### Possible Solution

I attempted to modify the source code by adding additional parameter inputs for the embedder in misc.py and embedder-base.py. However, when running the kickoff, I still encounter an error related to SSO.
`AuthenticationError: Error code: 401 - {'statusCode': 401, 'message': 'Invalid Authorization token'}
`

### Additional context

-

**Comment by voytas75** [comment]
first start with updated crewAI:

```python
pip install --upgrade crewai crewai-tools
```

[CrewAi now uses LiteLLM ](https://docs.crewai.com/how-to/LLM-Connections/?h=#supported-providers)

embedder for AOAI:

```python
    embedder={
        "provider": "azure_openai",
        "config":{
            "model": "<model>",
            "deployment_name": "<dep name>",
        },        
    },  
```

And  delete "OPENAI_API_BASE" env if exists: https://github.com/langchain-ai/langchain/discussions/17790#discussioncomment-8690960

**Comment by voytas75** [comment]
... and prepare env, example:

```
os.environ.update({
    "AZURE_API_KEY": os.getenv("AZURE_OPENAI_API_KEY"),
    "AZURE_API_BASE": os.getenv("AZURE_OPENAI_ENDPOINT"),
    "AZURE_API_VERSION": os.getenv("AZURE_OPENAI_API_VERSION"),
})
```

**Comment by morhyak** [comment]
Hi, thanks for your response.
However, the SSO issue wasn't addressed. I need to define the **Authorization tokens** as follows:"

` embedder={
        "provider": "azure_openai",
        "config": {
            "model": EMBEDDING_MODEL,
            "deployment": EMBEDDING_MODEL ,
            "openai_api_key":API_KEY,
            "azure_endpoint": API_BASE,
            "openai_api_version":API_VERSION,
            "openai_api_type":TYPE,
            "default_headers": {
                "**Authorization**": f"Bearer {access_token}",
                "**Content-Type**": "application/json"
            }
        }})`

**Comment by morhyak** [comment]
After configuring all the parameters, I encountered an error related to the SSO issue.

`AuthenticationError: Error code: 401 - {'statusCode': 401, 'message': 'Invalid Authorization token'}
`

**Comment by voytas75** [comment]
API_BASE is OK (https://YOUR_RESOURCE_NAME.openai.azure.com/)?  ```Invalid Authorization token``` probably come from OpenAI. Azure OpenAI has another text ```Access denied due to invalid subscription key. Make sure to provide a valid key for an active subscription. ```

**Comment by sorin-costea** [comment]
Shouldn't the Azure OpenAI base be like https://<zone>.api.cognitive.microsoft.com/ ?

**Comment by voytas75** [comment]
No, https://learn.microsoft.com/en-us/azure/ai-services/openai/reference and Azure Cognitive Services APIs is other service. I do not think crewai by itself or liteLLM support Cognitive.

**Comment by morhyak** [comment]
All the parameter configurations are correct (I double-checked them in another process). However, after adding extra options to the parameters in the config(for the sso tokens) , I am now encountering the following error:

`ConnectError: [Errno 11001] getaddrinfo failed
`

**Comment by sorin-costea** [comment]
> No, https://learn.microsoft.com/en-us/azure/ai-services/openai/reference and Azure Cognitive Services APIs is other service. I do not think crewai by itself or liteLLM support Cognitive.

LiteLLM says it supports the cognitive endpoints already (see https://github.com/BerriAI/litellm/discussions/5995), and (as far as I can tell) the only way to use Azure OpenAI is with this Cognitive regional service endpoint. The resource-based endpoints are called legacy and less and less regions support them. I wasn't able to find a region supporting them but also didn't try with every region...

**Comment by voytas75** [comment]
ok, maybe they are cooking :)
In their docs https://docs.litellm.ai/docs/providers/azure there is no info about ...cognitive.microsoft.com...
so does your code using cognitive.microsoft.com work on LiteLLM?
